File: backend/src/main/java/com/ndoruhirwe/smartlogistics/dto/response/GoodsReceptionResponse.java
package com.ndoruhirwe.smartlogistics.dto.response;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record GoodsReceptionResponse(

        UUID id,

        String reference,

        UUID purchaseOrderId,

        String purchaseOrderNumber,

        UUID purchaseOrderItemId,

        UUID productId,

        String productSku,

        String productName,

        UUID warehouseId,

        String warehouseCode,

        String warehouseName,

        BigDecimal quantity,

        String notes,

        LocalDateTime receptionDate,

        String createdBy,

        LocalDateTime createdAt
) {
}

File: frontend/src/app/features/procurement/models/goods-reception-response.ts
export interface GoodsReceptionResponse {
  id: string;
  reference: string;

  purchaseOrderId: string;
  purchaseOrderNumber: string;

  purchaseOrderItemId: string;

  productId: string;
  productSku: string;
  productName: string;

  warehouseId: string;
  warehouseCode: string;
  warehouseName: string;

  quantity: number;

  notes: string | null;

  receptionDate: string;
  createdBy: string;
  createdAt: string;
}
